### ISSUE TITLE
What is class supervision?!

### ISSUE THREAD

**Issue by AdityaNikhil** [PeterouZh/Omni-GAN-PyTorch#1]
Hi there,

I just completed reading your paper, and I see that in the paper you were stressing on class supervision a lot. Also in sec 3.3, you mentioned that, **notable performance of Omni-GAN comes from making full use of class supervision.** 

Can you explain what is class supervision here?!

**Comment by PeterouZh** [comment]
Hi. Thanks for your interest in our work!

"Class supervision" refers to the class label of an image. In fact, both projection-based loss and omni-loss employ class labels. But projection-based loss is a weak implicit classification loss. Omni-loss is a classification loss that makes full use of the class label's supervision. Thanks for asking this. We will clarify it in an updated version of the paper.

We have done classification experiments to show that the classifier's classification ability trained with projection-based loss is relatively weak. One way to enhance the classification ability of projection-based loss is to increase the batch size. This also explains why using a large batch size for BigGAN can improve the performance. We will update the manuscript as soon as possible to detail this experiment.

On the other hand, a strong classification loss will cause the discriminator to overfit the training data, which will cause the training of cGANs to collapse earlier. As mentioned in the paper, using weight decay on the discriminator can effectively solve this problem. 

The conclusion is that for cGANs, using a strong classification loss function combined with an appropriate regularization method (e.g., weight decay employed in the paper) for the discriminator can help achieve significant performance improvements. 

Recently I have access to large-scale GPU resources. Experiments show that Omni-GAN has completely surpassed BigGAN in the ImageNet 128x128 experiment. Besides, there is no need for a huge batch size like BigGAN (256 vs. 2048 for Omni-GAN and BigGAN respectively). These experiments will be added to our manuscript.

**Comment by AdityaNikhil** [comment]
Thank you for the explanation.!
I just have one more question, i searched all over internet but couldn't get proper answer.
Can you explain the major differences between projection based and classification based loss functions?!

**Comment by PeterouZh** [comment]
![image](https://user-images.githubusercontent.com/26176709/104691235-cd5ed600-5740-11eb-8926-03914d3e121b.png)

Suppose there is a real image (x_real, y_real) and a generated image (x_fake, y_fake). y is the corresponding class label of x. 

Note that the original projection-based discriminator is implemented in the manner of class embedding (refer to the [code](https://github.com/ajbrock/BigGAN-PyTorch/blob/98459431a5d618d644d54cd1e9fceb1e5045648d/BigGAN.py#L403)). This can be transformed into a discriminator whose last layer is a multi-dimensional fully connected layer. As shown above, the discriminator takes as input a image and outputs a (C+1)-dimensional vector, where C is the number of classes. 

The main difference between projection-based loss and classification-based loss is that projection-based loss is a weaker implicit classification loss (In one iteration, there are many logits that are not optimized, e.g., s_0, s_1, ..., s_{c-1}). However, for classification-based loss, all logits will be optimized. 

Note that for the sake of simplicity in the above figure, the output dimension of the discriminator is C+1 dimension, and its function is similar to the C+2 dimension used in our paper. In our paper, in order to decouple the real and fake categories of GAN, one more dimension is used. 

Hope this helps!
